### DISCUSSION TITLE
Thanks - I've borrowed your code for grouping options into tabs

### DISCUSSION THREAD

**Discussion by ronzulu** [Taitava/obsidian-shellcommands#421] [Show and tell]
Hi @Taitava 

Thanks for the idea and implementation for grouping options into tabs. On the `spaced repetition` plugin we have so many options, and was probably unworkable for some users.
https://github.com/st3v3nmw/obsidian-spaced-repetition

Now looks like...
![image](https://github.com/user-attachments/assets/aef784fa-890e-4a5f-80e0-665a00263ec5)

Cheers
Ronny

**Comment by Taitava** [comment]
Hi @ronzulu ! Thank you for letting me know this! 🙂 Nice that the tabs code works well for your plugin, too!

Do you think it would make sense for me to create a separate GitHub repository and NPM package for the tabs component?
- I could improve the tabs API to make it easier to use.
- It would be easier for Obsidian community plugins to pull updates for the tabs component (if wanted).
- There would be a dedicated place to report problems/ideas. I would be willing to receive pull requests, too. 🙂

**Comment by ronzulu** [comment]
Hi @Taitava great idea.

If you include some documentation, it might be helpful if you mention about the need to rename the CSS classes. And maybe start off calling them ".xxx-tab-content" etc.

I only have some experience with CSS and it took me a while to realise that the CSS class names are global in scope. So at one time the spaced repetition plug-in worked, but only when your plug-in was also installed!

Cheers
Ronny

**Reply by Taitava** [reply]
Good point! I need to write to the documentation that all CSS classes need prefixing, i.e. each plugin using the tabs should use their own prefix. As you have noticed, for the _Shell commands_ plugin, the prefix is `SC-`:
https://github.com/Taitava/obsidian-shellcommands/blob/bfe508eeecdc5b31d533e9ff3f5e428a3fe5c079/styles.css#L239-L287

---

The documentation also needs to emphasize that CSS needs to be manually put into the plugin's own `styles.css` file, so just an `npm install` is not enough.
